posture/serialnumber_macos.go

```diff
@@ -0,0 +1,40 @@
+// Copyright (c) Tailscale Inc & AUTHORS
+// SPDX-License-Identifier: BSD-3-Clause
+
+//go:build darwin
+
+package posture
+
+// #cgo LDFLAGS: -framework CoreFoundation -framework IOKit
+// #include <CoreFoundation/CoreFoundation.h>
+// #include <IOKit/IOKitLib.h>
+//
+// const char *
+// getSerialNumber()
```

Comment:
Is there any error handling we can add here?

Reading the docs and looking at usage of these functions in other code on Github, I believe:
- `IOServiceMatching` can return zero on failure
- `IOServiceGetMatchingService` might return zero if service could not be matched
- `IORegistryEntryCreateCFProperty` could return zero too
- `CFStringGetCStringPtr` can return NULL if string cannot be extracted with no memory allocations or copying, in which case [the docs](https://developer.apple.com/documentation/corefoundation/1542133-cfstringgetcstringptr) recommend using `CFStringGetCString` instead.

Reply:
will give it a go

Reply:
Added error handling.